libbeat/docs/upgrading.asciidoc

```diff
@@ -4,214 +4,142 @@
+Elasticsearch and Kibana, and running Beats 6.0 against Elasticsearch/Kibana
+version 5.6 is supported.
 
-Beats 5.0 comes with several backwards incompatible configuration changes.
-However, we provide a script that can automatically fixup the most common
-settings in your configuration files.
+If you upgrade Elasticsearch before Beats, it's important to apply the 5.6
+template before starting the Elasticsearch upgrade. So if you are on a 5.x
+version lower than 5.6, please follow the <<upgrading-to-5.6>>
+section *before* doing the Elasticsearch upgrade.
 
-You can find this script in the Beats download packages. For the `.tar.gz` and
-`.zip` archives, look under:
+[[upgrading-to-5.6]]
+==== Upgrade to 5.6
 
-[source,shell]
-------------------------------------------------------------------------------
-./scripts/migrate_beat_config_1_x_to_5_0.py
-------------------------------------------------------------------------------
-
-For the `.deb` and `.rpm` packages, look under the
-`/usr/share/<beatname>/scripts/` folder, for example:
+The upgrade procedure assumes that you have Beats version 5.6 installed. If you
+are on a previous 5.x version of the Beats, please upgrade to version 5.6 first.
+The reason is that the Elasticsearch mapping template in 5.6 was modified to be
+compatible with Elasticsearch 6.0 (by removing the `_all` settings).
 
+For this reason, after upgrading to 5.6, you need to make sure that the 5.6
+template is loaded. You can do this by temporarily enabling the
+`output.elasticsearch.template.overwrite` setting, for example with Metricbeat:
 
 [source,shell]
 ------------------------------------------------------------------------------
-/usr/share/packetbeat/scripts/migrate_beat_config_1_x_to_5_0.py
+metricbeat -e -E output.elasticsearch.template.overwrite=true
 ------------------------------------------------------------------------------
 
-Python 2.7 or later is required for executing the migration script.
-
-To upgrade your configuration file automatically, simply pass it as an argument
-to the migration script, for example:
+Alternatively, you can manually force loading the template:
 
 [source,shell]
 ------------------------------------------------------------------------------
-> ./scripts/migrate_beat_config_1_x_to_5_0.py packetbeat.yml
-Backup file created: packetbeat.yml.bak
+curl -XPUT -H'Content-Type: application/json'  http://localhost:9200/_template/metricbeat -d @metricbeat.template.json
 ------------------------------------------------------------------------------
 
-The script will automatically create a backup file and overwrite the input file
-with the upgraded version. You can also use the `--dry` option, which doesn't
-modify the input file, but outputs the upgraded version to stdout instead.
+To check which version of the template is loaded, open Kibana Console, call `GET
+/_template/metricbeat`, and look for the version string. Note that you need to
```

Comment:
why not using the `export` command for that? In Beats 6.0, it's `./metricbeat export template`.

Reply:
Same as above, we're still on 5.6 at that step.